ffmpeg/lpms_ffmpeg.c

```diff
@@ -1148,11 +1177,17 @@ int process_out(struct input_ctx *ictx, struct output_ctx *octx, AVCodecContext
   // Start filter flushing process if necessary
   if (!inf && !filter->flushed) {
     // Set input frame to the last frame
-    // And increment pts offset by pkt_duration
-    // TODO It may make sense to use the expected output packet duration instead
     int is_video = AVMEDIA_TYPE_VIDEO == ost->codecpar->codec_type;
     AVFrame *frame = is_video ? ictx->last_frame_v : ictx->last_frame_a;
-    filter->flush_offset += frame->pkt_duration;
+    // Increment pts offset by:
+    // - input packet's duration usually
+    int64_t dur = frame->pkt_duration;
+    // - output packet's duration if using fps filter & haven't encoded anything yet
+    if (is_video && octx->fps.den && !octx->res->frames) {
+      AVStream *ist = ictx->ic->streams[ictx->vi];
+      dur = av_rescale_q(frame->pkt_duration, ist->r_frame_rate, octx->fps);
+    }
+    filter->flush_offset += dur;
```

Comment:
Here we increment `filter->flush_count += frame->pkt_duration` ; do we have to make a similar adjustment there as well?

https://github.com/livepeer/lpms/blob/42a805ec2debbfa6b448a28b16650f4e186dcb5e/ffmpeg/lpms_ffmpeg.c#L1229

Reply:
Setting it to `dur` instead of keeping it as-is lead to some errors in some testcases, I'll check again to be sure.

Reply:
Ah it is a no-op, as we are rescaling duration only when fps filter is used, and line 1229 is never executed in that case.